features/reward_features.py

```diff
@@ -291,6 +291,55 @@ def _start_tracking_out(self):
             super()._start_tracking_out()
             self.reward.off()
 
+
+class ScoreRewards(traits.HasTraits):
+    '''
+    Add a "score" to the task that awards points based on target acquisition speed. 
+    The score is displayed after each reward and on the web GUI. The running score also gets
+    saved as a value in the task data called 'reward_score'.
+
+    Note:
+        Only works with target acquisition tasks.
+    '''
+    score_display_location = traits.Tuple((10, 0, 10), desc="Location to display the score (in cm)")
+    score_display_height = traits.Float(1, desc="Height of the score display (in cm)")
+    score_display_color = traits.Tuple((1, 1, 1, 1), desc="Color of the score display")
+    score_timed_state = traits.String("target", desc="State to display the score after")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.reportstats['Score'] = 0
+    
+    def init(self):
+        self.add_dtype('reward_score', 'f8', (1,))
+        super().init()
+        self.task_data['reward_score'] = 0
+
+    def _start_reward(self):
+        if hasattr(super(), '_start_reward'):
+            super()._start_reward()
+        timed_state = None
+        idx = -1
+        while timed_state is None and -idx-1 < len(self.state_log):
+            if self.state_log[idx][0] == self.score_timed_state:
+                timed_state = self.state_log[-1][1] - self.state_log[idx][1]
+            idx -= 1
+        if timed_state is None or timed_state == 0.:
+            score = 0.
+        else:
+            score = 10*int(10./timed_state)
+        self.reportstats['Score'] += score
+        self.score_display = TextTarget(str(score), height=self.score_display_height, 
+                                        color=self.score_display_color, background_color=self.background)
+        self.score_display.move_to_position(self.score_display_location)
+        self.add_model(self.score_display.model)
+        self.task_data['reward_score'] += score
```

Comment:
Is there a reason you use `+=` instead of `=`?

Reply:
Not really, choosing between having to integrate to get the sum or take a gradient to get the individual scores. Both are easy enough, I just went with keeping track of the cumulative score and having to take a gradient for the individual trials.